### PR TITLE
Bug 1907828: Adding pvc health checking when pod is pending

### PR DESCRIPTION
**Pull request by shawn-hurley**
* not just init container

### CODE REVIEW

**Review by jmontleon — APPROVED** [review]
ACK

**Review by eriknelson — APPROVED** [review]

**Review by alaypatel07 — APPROVED** [review]
ack

File: pkg/controller/migmigration/stage.go [hunk 1]
```diff
@@ -534,6 +534,14 @@ func (t *Task) stagePodReport(client k8sclient.Client) (report PodStartReport, e
 
 		// handle pod pending status
 		if pod.Status.Phase == corev1.PodPending {
+			// If the pod has unhealthy claims, we will fail the migration
+			// So the user can fix the plan/pvc.
+			// pod Spec having the node name, means the pod has been scheduled
+			// becuase the pod has been scheduled we know that the PVC should be bound
+			if pod.Spec.NodeName != "" && !hasHealthyClaims(&pod) {
```
Comment (alaypatel07) [inline pkg/controller/migmigration/stage.go:541]: understanding here is that if NodeName == "", we wait for the pod to be scheduled. +1 coming from this test case https://github.com/kubernetes/kubernetes/blob/e1c1fd9edf9bd4d9d0e749b16fb33a54399a1ef8/test/integration/volumescheduling/volume_binding_test.go#L831

**Review by alaypatel07 — APPROVED** [review]
re-ack

**Pull request merged**